urllib3/exceptions.py

```diff
@@ -112,6 +112,9 @@ class ConnectTimeoutError(TimeoutError):
     "Raised when a socket timeout occurs while connecting to a server"
     pass
 
+class NewConnectionError(ConnectTimeoutError, PoolError):
```

Comment:
Yea, maybe I'll cut PoolError. They're mostly supposed to be for pool state violations (ie. if you fiddle with internal mechanics and ruin a counter somewhere).